biomechanics: Minor Fixes math rendering and added hyperlinks

bsach64 opened the pull request:
Fixes #25785
Fixed math rendering of:
- FiberForceLengthActiveDeGroote2016
- FiberForceLengthPassiveDeGroote2016 
Also added sphinx referencing for with_defaults in:
- FiberForceLengthActiveDeGroote2016
- FiberForceLengthPassiveDeGroote2016

<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
Fixes #25785 
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->


#### Brief description of what is fixed or changed


#### Other comments


#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

bsach64 committed:
biomechanics: Minor Fixes to docs

Fixes #25785
Fixed math rendering of:
- FiberForceLengthActiveDeGroote2016
- FiberForceLengthPassiveDeGroote2016
Also added sphinx referencing for with_defaults in:
- FiberForceLengthActiveDeGroote2016
- FiberForceLengthPassiveDeGroote2016

bsach64 commented:
FiberForceLengthActiveDeGroote2016 renders as:
![image](https://github.com/sympy/sympy/assets/69461338/876427d9-cef1-4504-93c5-4d9760b69e12)
FiberForceVelocityDeGroote2016 renders as: 
![image](https://github.com/sympy/sympy/assets/69461338/461f6370-0982-448e-9bfd-6393544b3c86)

bsach64 committed:
Adds sphinx referencing for functions

Added sphinx referencing in examples for with_defaults and doit in:
- TendonForceLengthDeGroote2016
- TendonForceLengthInverseDeGroote2016
- FiberForceLengthPassiveDeGroote2016
- FiberForceLengthPassiveInverseDeGroote2016
- FiberForceLengthActiveDeGroote2016
- FiberForceVelocityDeGroote2016
- FiberForceVelocityInverseDeGroote2016

brocksam commented on sympy/physics/biomechanics/curve.py:
```diff
@@ -959,9 +959,9 @@ class FiberForceLengthActiveDeGroote2016(CharacteristicCurveFunction):
-    $fl^M_{act} = c_0 \exp{-\frac{1}{2}\left(\frac{\tilde{l}^M - c_1}{\left(c_2 + c_3 \tilde{l}^M\right)}\right)^2}
-        + c_4 \exp{-\frac{1}{2}\left(\frac{\tilde{l}^M - c_5}{\left(c_6 + c_7 \tilde{l}^M\right)}\right)^2}
-        + c_8 \exp{-\frac{1}{2}\left(\frac{\tilde{l}^M - c_9}{\left(c_{10} + c_{11} \tilde{l}^M\right)}\right)^2}$
+    $f_{\text{act}}^M = c_0 \exp\left(-\frac{1}{2}\left(\frac{\tilde{l}^M - c_1}{c_2 + c_3 \tilde{l}^M}\right)^2\right)
```
```suggestion
    $fl_{\text{act}}^M = c_0 \exp\left(-\frac{1}{2}\left(\frac{\tilde{l}^M - c_1}{c_2 + c_3 \tilde{l}^M}\right)^2\right)
```

brocksam committed:
Fix typo in symbol name

brocksam commented:
Thanks very much for the contribution!

@moorepants: do you have any opinions on the use of Sphinx directives, e.g. `:meth:`? They improve the online docs but clutter the command line docs. I avoided using them because I didn't see them used anywhere else in SymPy. If we're happy for them to be used then this PR is ready for merging. The docs would also benefit from using them more extensively throughout.

oscarbenjamin commented:
> I avoided using them because I didn't see them used anywhere else in SymPy.

They are used in lots of places:
```
$ git grep :meth: | wc -l
350
$ git grep :func: | wc -l
416
$ git grep :class: | wc -l
1127
```
I always use them because I want hyperlinked references. Often the most useful thing that a docstring can do is to link to other places.

moorepants commented:
It is true that they are helpful in html but not in plain text. That goes for other sphinx/rst markup too. I use plaintext docs myself significantly more than html docs, but I'm ambivalent and I requested they be added.

brocksam commented:
@bsach64 are you also able to add other relevant Sphinx directives to this PR? E.g. `:class:` where we have `TendonForceLengthDeGroote2016`. Basically anywhere we have double back ticks.

bsach64 commented:
> @bsach64 are you also able to add other relevant Sphinx directives to this PR? E.g. `:class:` where we have `TendonForceLengthDeGroote2016`. Basically anywhere we have double back ticks.

Sure!

bsach64 committed:
Adds Sphinx Referencing for classes

Added spinx referencing for Symbol and class itself:
- TendonForceLengthDeGroote2016
- TendonForceLengthInverseDeGroote2016
- FiberForceLengthPassiveDeGroote2016
- FiberForceLengthPassiveInverseDeGroote2016
- FiberForceLengthActiveDeGroote2016
- FiberForceVelocityDeGroote2016
- FiberForceVelocityInverseDeGroote2016

bsach64 committed:
Removed an ` that caused docs build to fail
